### PR TITLE
perf_script_bug.py:Testcase for perf script segfault

### PR DESCRIPTION
**Pull request by shirishaganta**
Adding testcase for perf script seg fault
with cpu cycle related events

Signed-off-by: Shirisha Ganta <shiganta@in.ibm.com>

### CODE REVIEW

**Comment by shirishaganta** [comment]
Fail case:
=====================
JOB ID     : a90973a80d8563c33dea045683d6338d2a684427
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-09-11T02.58-a90973a/job.log
 (1/1) test.py:sample.test_probe: FAIL: problem with perf script (0.71 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1.15 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-09-11T02.58-a90973a/results.html

pass case:
==============
JOB ID     : 47d5062f3c67d6e1f9b955bd65faf275f4b37b8d
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-09-11T02.20-47d5062/job.log
 (1/1) test.py:sample.test_probe: PASS (2.99 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 3.53 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-09-11T02.20-47d5062/results.html

File: perf/perf_script_bug.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from avocado.utils.software_manager import SoftwareManager
+
+
+class PerfProbe(Test):
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf
+        '''
+
```
Comment (PraveenPenguin) [inline perf/perf_script_bug.py]: please remove this white space

File: perf/perf_script_bug.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+from avocado import main
+from avocado.utils import build, distro, process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class PerfProbe(Test):
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf
```
Comment (PraveenPenguin) [inline perf/perf_script_bug.py]: is this like 

`Install basic packages to support perfProbe test`

File: perf/perf_script_bug.py (hunk 1 above)
Comment (PraveenPenguin) [inline perf/perf_script_bug.py]: please remove this white space

File: perf/perf_script_bug.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.distro_name = detected_distro.name
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in self.distro_name:
+            deps.extend(['linux-tools-common', 'linux-tools-%s' %
+                         platform.uname()[2]])
+        elif self.distro_name in ['rhel', 'SuSE', 'fedora', 'centos']:
```
Comment (PraveenPenguin) [inline perf/perf_script_bug.py]: is this tested on fedora or centos ?

File: perf/perf_script_bug.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            deps.extend(['perf'])
+        else:
+            self.cancel("Install the package for perf supported\
```
Comment (PraveenPenguin) [inline perf/perf_script_bug.py]: this statement looks like confusing can you please rephrase it

File: perf/perf_script_bug.py.data/perf_test.c [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#include <stdio.h>
```
Comment (PraveenPenguin) [inline perf/perf_script_bug.py.data/perf_test.c]: author signature please

**Comment by shirishaganta** [comment]
@PraveenPenguin i have done all the changes which you requested above

**Pull request merged**